cmd/kubeadm/app/cmd/phases/preflight.go

```diff
@@ -45,17 +65,31 @@ type preflightMasterData interface {
 	IgnorePreflightErrors() sets.String
 }
 
+type preflightJoinData interface {
+	Cfg() *kubeadmapi.JoinConfiguration
+	InitCfg() (*kubeadmapi.InitConfiguration, error)
+	IgnorePreflightErrors() sets.String
+}
+
 // NewPreflightMasterPhase creates a kubeadm workflow phase that implements preflight checks for a new master node.
 func NewPreflightMasterPhase() workflow.Phase {
```

Comment:
Shouldn't we actually nuke this phase as we agreed in the kubeadm office hours meeting last week, that we need to combine all checks in one phase and that the checks here are now also performed in the pre-flight join phase?

Reply:
@MalloZup 
in this issue we have a list of sub-phases we need to call in this preflight phase:
https://github.com/kubernetes/kubeadm/issues/1204

let me know which items are unclear to you.
there is also a linked google doc with some comments next to it.

Reply:
@neolit123 @rosti thx now i get it. ok thx. i will update it asap. Thx :+1:

Reply:
@fabriziopandini cc

Reply:
@rosti If I remember well the discussion, it was agreed to merge all the various preflight checks implemented in the join workflow and detailed in @neolit123 document into a unique phase, that is `NewPreflightJoinPhase` as proposed in this PR. 

Instead, I don't think we should merge `NewPreflightJoinPhase` and `NewPreflightMasterPhase` because they do different things.

@neolit123 I don't see a use case for invoking only a subgroup of join preflight checks ("we should combine all checks into one [phase]", as noted in the document)...

Reply:
> Should we make any changes to the preflight master phase in this PR

the diff is confusing enough already.
this is for a separate PR, i've added a comment for @MalloZup to add a TODO for rename `master` -> `control`. we can also consider this refactor later.

Reply:
ok done

Reply:
hm, i don't see the TODO in the latest diff.
but it's OK....we can get back to this.